instrumentation-core/reactor-3.1/src/main/java/io/opentelemetry/instrumentation/reactor/TracingOperator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.instrumentation.reactor;
+
+import io.grpc.Context;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import org.reactivestreams.Publisher;
+import reactor.core.CoreSubscriber;
+import reactor.core.Fuseable;
+import reactor.core.Scannable;
+import reactor.core.publisher.Hooks;
+import reactor.core.publisher.Operators;
+
+/** @author Based on Spring Sleuth's Reactor instrumentation. */
+public class TracingOperator {
+
+  /**
+   * Registers a hook that applies to every operator, propagating {@link Context} to downstream
+   * callbacks to ensure spans in the {@link Context} are available throughout the lifetime of a
+   * reactive stream. This should generally be called in a static initializer block in your
+   * application.
+   */
+  public static void registerOnEachOperator() {
+    Hooks.onEachOperator(tracingLift());
+  }
+
+  /** Unregisters the hook registered by {@link #registerOnEachOperator()}. */
+  public static void resetOnEachOperator() {
+    Hooks.onEachOperator(tracingLift());
```

Comment:
```suggestion
    Hooks.resetOnEachOperator(tracingLift());
```

Can you add a test for reset?